### PR TITLE
LPS-85386 dynamic-data-lists-test: Add integration test to check the indexed fields and reindexing for the DDLRecord and DDLRecordSet.

### PR DESCRIPTION
**Pull request by brandizzi**

### CODE REVIEW

**Comment by leomiyagi** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by leomiyagi** [comment]
Reviewing <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/search/test/DDLRecordSetFixture.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+public class DDLRecordSetFixture {
+
+	public DDLRecordSetFixture(Group group, User user) throws Exception {
+		_group = group;
+		_user = user;
+
+		setUpDdlRecordTestHelper();
+	}
+
+	public DDLRecordSet createAnDDLRecordSet() throws Exception {
```
Comment (leomiyagi) [inline modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/search/test/DDLRecordSetFixture.java]: Little typo here. Suggestion "createDDLRecordSet"

**Comment by leomiyagi** [comment]
@brandizzi does this replaces also brandizzi#666 ?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceIndexerIndexedFieldsTest.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+ */
+@RunWith(Arquillian.class)
+public class DDMFormInstanceIndexerIndexedFieldsTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE,
```
Comment (leomiyagi) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceIndexerIndexedFieldsTest.java]: This needs to be changed to "PermissionCheckerMethodTestRule". Make the change on all test classes.

**Comment by leomiyagi** [comment]
ci:test:search

**Comment by leomiyagi** [comment]
ci:sf:test:search:norebase

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceFixture.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ * @author Lucas Marques
+ */
+public class DDMFormInstanceFixture {
+
+	public DDMFormInstanceFixture(Group group, User user) {
+		_group = group;
+		_user = user;
+	}
+
+	public DDMFormInstance createAnDDMFormInstance() throws Exception {
```
[the remark this answers is not in the document]
Comment (vagnerbc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceFixture.java]: ok!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceFixture.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+	public void updateDisplaySettings(Locale locale) throws Exception {
+		Group group = GroupTestUtil.updateDisplaySettings(
+			_group.getGroupId(), null, locale);
+
+		_group.setModelAttributes(group.getModelAttributes());
+	}
+
+	protected DDMFormInstance addDDMFormInstance() throws Exception {
+		DDMFormInstanceTestHelper ddmFormInstanceTestHelper =
```
[the remark this answers is not in the document]
Comment (vagnerbc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceFixture.java]: ok!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceIndexerIndexedFieldsTest.java [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+		group = userSearchFixture.addGroup();
+
+		user = userSearchFixture.addUser(RandomTestUtil.randomString(), group);
+	}
+
+	protected DDMFormInstanceFixture ddmFormInstanceFixture;
+	protected IndexerFixture<DDMFormInstance> ddmFormInstanceIndexerFixture;
+
+	@Inject
+	protected DDMIndexer ddmIndexer;
```
[the remark this answers is not in the document]
Comment (vagnerbc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceIndexerIndexedFieldsTest.java]: ok!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceRecordIndexerReindexTest.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+	protected void setUpDDMFormInstanceRecordFixture() throws Exception {
+		ddmFormInstanceRecordFixture = new DDMFormInstanceRecordFixture(
+			group, user);
+
+		_ddmFormInstanceRecords =
+			ddmFormInstanceRecordFixture.getDdmFormInstanceRecords();
+	}
+
+	protected void setUpUserIndexerFixture() {
```
[the remark this answers is not in the document]
Comment (vagnerbc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceRecordIndexerReindexTest.java]: ok!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceIndexerReindexTest.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+		user = userSearchFixture.addUser(RandomTestUtil.randomString(), group);
+	}
+
+	protected DDMFormInstanceFixture ddmFormInstanceFixture;
+	protected IndexerFixture<DDMFormInstance> ddmFormInstanceIndexerFixture;
+	protected Group group;
+
+	@Inject
+	protected SearchEngineHelper searchEngineHelper;
```
[the remark this answers is not in the document]
Comment (vagnerbc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceIndexerReindexTest.java]: ok!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceRecordFixture.java [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+	public DDMStructure getDdmStructure() {
+		return _ddmStructure;
+	}
+
+	public ServiceContext getServiceContext() throws Exception {
+		return ServiceContextTestUtil.getServiceContext(
+			_group.getGroupId(), _user.getUserId());
+	}
+
+	protected DDMFormInstanceRecord addDDMFormInstanceRecord(
```
[the remark this answers is not in the document]
Comment (vagnerbc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceRecordFixture.java]: ok!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceRecordIndexerIndexedFieldsTest.java [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+
+		return indexableAttributes.trim();
+	}
+
+	private void _populateAttributes(
+			DDMFormInstanceRecord ddmFormInstanceRecord,
+			Map<String, String> map)
+		throws Exception {
+
+		Document document = new DocumentImpl();
```
[the remark this answers is not in the document]
Comment (vagnerbc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceRecordIndexerIndexedFieldsTest.java]: ok!

File: modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/search/test/DDLRecordFixture.java [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+	}
+
+	public void updateDisplaySettings(Locale locale) throws Exception {
+		Group group = GroupTestUtil.updateDisplaySettings(
+			_group.getGroupId(), null, locale);
+
+		_group.setModelAttributes(group.getModelAttributes());
+	}
+
+	protected DDLRecord addRecord(
```
[the remark this answers is not in the document]
Comment (vagnerbc) [reply modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/search/test/DDLRecordFixture.java]: ok!

File: modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/search/test/DDLRecordSetFixture.java (hunk 1 above)
Comment (vagnerbc) [reply modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/search/test/DDLRecordSetFixture.java]: ok!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceIndexerIndexedFieldsTest.java (hunk 2 above)
Comment (vagnerbc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/search/test/DDMFormInstanceIndexerIndexedFieldsTest.java]: ok!

**Comment by vagnerbc** [comment]
Changes updated

**Comment by leomiyagi** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by leomiyagi** [comment]
ci:test:sf

**Comment by leomiyagi** [comment]
Pull request submitted to https://github.com/brandizzi/liferay-portal/pull/772. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**